cmd/erasure-object.go

```diff
@@ -545,6 +559,46 @@ func (er erasureObjects) deleteIfDangling(ctx context.Context, bucket, object st
 	return m, err
 }
 
+func (er erasureObjects) deleteBucketIfDangling(ctx context.Context, bucket string, metaArr []VolInfo, errs []error) error {
+	_, file, line, cok := runtime.Caller(1)
+	var err error
+	ok := isBucketDangling(metaArr, errs)
+	if ok {
+		tags := make(map[string]interface{}, 4)
+		tags["set"] = er.setIndex
+		tags["pool"] = er.poolIndex
+		tags["merrs"] = joinErrors(errs...) // errors.Join(errs...)
+
+		if cok {
+			tags["caller"] = fmt.Sprintf("%s:%d", file, line)
+		}
+
+		defer auditDanglingBucketDeletion(ctx, bucket, tags)
+
+		err = errVolumeNotFound
+
+		disks := er.getDisks()
+		g := errgroup.WithNErrs(len(disks))
+		for index := range disks {
+			index := index
+			g.Go(func() error {
+				if disks[index] == nil {
+					return errDiskNotFound
+				}
+				err := disks[index].DeleteVol(ctx, bucket, false)
+				if errors.Is(err, errVolumeNotEmpty) {
+					// If volume non empty, force delete as its dangling anyway
+					return disks[index].DeleteVol(ctx, bucket, true)
```

Comment:
I am really not in favor of doing Force Delete here; the benefit is low comparing to the risk

Reply:
ok, so no need to have force delete called at all? will remove.

Reply:
My suggestion is @shtripat let's leave out the force part instead just perform normal DelVol if possible if its not they engage support and we investigate.

Keeping it manual is safer.

Reply:
ok, I will remove the force delete part here, but in actual dangling bucket scenario where objects as well left out, this code wont have any effect as bucket wont be deleted due to non empty. We can just log an error once that during removal of dangling buckets, it was not able to delete as bucket was non empty.